server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4377,6 +4377,9 @@ public void close() throws IOException {
      * @throws IOException if exception occurs while reading segments from remote store
      */
     public void syncSegmentsFromRemoteSegmentStore(boolean overrideLocal) throws IOException {
+        if (recoveryState.getStage() != RecoveryState.Stage.INDEX) {
```

Comment:
nit - should we add an assert here and control this from the caller itself; what I mean is that calling this method when recovery state stage is not Index is probably not the correct thing to do in the first place. By the looks, for the caller it could lead to unexpected behaviour while this call is supposed to sync segments from remote store.

Reply:
Agree, I am not very much convinced on adding this block here (also the part of quietly returning). In SegRep integration with remote store, this can create issues (cc: @ankitkala ). Let me think more on this.